sdk/src/main/java/ch/admin/bag/covidcertificate/eval/net/JwsInterceptor.kt

```diff
@@ -10,76 +10,65 @@
+class JwsInterceptor(rootCA: X509Certificate) : Interceptor {
+
+	companion object {
+		// application/json media type for the payload of a JWS
+		private val APPLICATION_JSON = "application/json".toMediaType()
+	}
 
-    @Throws(IOException::class)
+	private val keyResolver: JwsKeyResolver = JwsKeyResolver(rootCA)
 
-    override fun intercept(chain: Interceptor.Chain): Response {
-        val response: Response = chain.proceed(chain.request())
-        if (!response.isSuccessful) {
-            return response
-        }
-        if (response.cacheResponse != null) {
-            return response
-        }
-        val jws = response.body?.string() ?: throw SignatureException("Body has no signature")
+	@Throws(IOException::class)
+	override fun intercept(chain: Interceptor.Chain): Response {
+		val response: Response = chain.proceed(chain.request())
+		if (!response.isSuccessful) {
+			return response
+		}
 
-        var body = ""
-        try {
-            val claimsJws : Jws<Claims> = Jwts.parserBuilder()
-                .setSigningKeyResolver(keyResolver)
-                    .build()
-                    .parseClaimsJws(jws)
-            // now that the JWS is verified, we can safely assume that the body can be trusted, so serialize it to JSON again
-            body = Moshi.Builder().build().adapter<Claims>(Types.newParameterizedType(Map::class.java, String::class.java, Object::class.java)).toJson(claimsJws.body)
-        } catch (e: io.jsonwebtoken.security.SignatureException) {
-            throw SignatureException("Invalid Signature")
-        } catch (o: ExpiredJwtException) {
-            throw SignatureException("Expired JWT")
-        }
+		val jws = response.body?.string() ?: throw SignatureException("Body has no signature")
+		val body: String
+		try {
+			val claimsJws: Jws<Claims> = Jwts.parserBuilder()
+				.setSigningKeyResolver(keyResolver)
+				.build()
+				.parseClaimsJws(jws)
+			// now that the JWS is verified, we can safely assume that the body can be trusted, so serialize it to JSON again
+			body = Moshi.Builder().build()
+				.adapter<Claims>(Types.newParameterizedType(Map::class.java, String::class.java, Object::class.java))
+				.toJson(claimsJws.body)
+		} catch (e: io.jsonwebtoken.security.SignatureException) {
+			throw SignatureException("Invalid Signature")
+		} catch (o: ExpiredJwtException) {
+			throw SignatureException("Expired JWT")
+		}
 
-        // SAFE ZONE
-        // from here on body contains a JSON-string of the payload of the JWS, whose signature we verified.
+		// SAFE ZONE
+		// from here on body contains a JSON-string of the payload of the JWS, whose signature we verified.
 
-        return response.newBuilder()
-                .body(
-                        body.toResponseBody(APPLICATION_JSON)
-                ).build()
-    }
+		return response.newBuilder()
+			.body(
+				body.toResponseBody(APPLICATION_JSON)
+			).build()
+	}
```

Comment:
This could probably also be squashed onto a single line (or at least 3 lines), but that's being pedantic about code you didn't write :)